src/game_inventory.cpp

```diff
@@ -535,21 +535,34 @@ class comestible_inventory_preset : public inventory_selector_preset
             }, _( "VOLUME" ) );
 
             append_cell( [&p]( const item_location & loc ) {
+                /* Understanding how Calories Per Effective Volume are calculated requires a dive into the
+                stomach fullness source code. Look at issue #44365*/
                 const item &it = *loc;
-                const int charges = std::max( it.charges, 1 );
-                const double converted_weight = convert_weight( it.weight() / charges );
-                if( converted_weight == 0 ) {
-                    return std::string( "---" );
-                }
-                const nutrients nutr = p.compute_effective_nutrients( *loc );
+                const nutrients nutr = p.compute_effective_nutrients( it );
                 const int kcalories = nutr.kcal;
-                // Experimental: if calories are 0 (medicine, batteries etc), don't display anything.
-                if( kcalories == 0 ) {
+                if( kcalories == 0 ||
+                    !it.type->comestible ) { //quit prematurely if there is no caloric content/the item is not food.
                     return std::string();
                 }
-                const int calpergr = int( std::round( kcalories / converted_weight ) );
-                return string_format( _( "%d" ), calpergr );
-            }, _( "CAL/kg" ) );
+                units::volume water_vol = ( it.type->comestible->quench > 0 ) ? it.type->comestible->quench *
+                                          5_ml : 0_ml;
+                units::volume food_vol = it.volume() - water_vol; //water volume is ignored.
+                const double converted_volume = round_up( convert_volume( food_vol.value() / it.count() ), 2 );
+                const double energy_density_ratio = p.compute_effective_food_volume_ratio( it );
+                const double effective_volume = converted_volume * energy_density_ratio;
+                const int calories_per_effective_volume = std::round( kcalories / effective_volume );
+                if( p.has_bionic(
+                        bionic_id( "bio_digestion" ) ) ||
+                    p.has_trait( trait_id( "SELFAWARE" ) ) ) { // change to also have a skill condition??
+                    return string_format( _( "%d" ), calories_per_effective_volume ); //return exact value
+                } //else compute "vague" values to display.
```

Comment:
I am not a fan of extending the random checks for specific bionics and traits, and I don't think this value should ever really be displayed numerically.
Also, please put comments on a line before or after lines, not on the same line, as doing so makes them really ugly when styled. 
```suggestion
```

Reply:
@anothersimulacrum: Why are you saying the checks are random? It gives both the bionic and the trait a use that fits both the bionic (digestion) and trait (it's about awareness of your body, after all)

Reply:
Yes, I do think that there should be _some_ way for the player to see the raw number.

Reply:
Actually, I think I will limit this PR to vague values. I will submit a new PR, once this one is merged, that adds support for the raw numbers, given certain conditions. This way, we can have a broader conversation on whether the raw numbers should be shown.

Reply:
It should not be displayed at all, it depends on a value that also should not be shown to the player, the effective stomach volume of the food.
They are not random because they what they are is unrelated to what the check does, but they are random in how they are used to extend behavior, because they are randomly strewn about the codebase without any sort of rules.
They are hardcoded, poorly maintainable, and make what things do harder to read and understand. This is not a pattern we should be extending.

Reply:
Keep in a check for `debug_mode` ? Would allow tracking down any food weirdness later on significantly easier.